src/KafkaNetClient/KafkaTcpSocket.cs

```diff
@@ -389,6 +421,10 @@ private static Exception FindExceptionInLastReadOrWriteTasks(Task lastWriteTask,
                 }
 
                 await Task.Delay(TimeSpan.FromMilliseconds(reconnectionDelay), _disposeToken.Token).ConfigureAwait(false);
+                if (_maxRetry < attempts)
```

Comment:
Move this condition inside the catch so we don't accidentally throw an exception if we succeeded.